### ISSUE TITLE
survClust

### ISSUE THREAD

**Issue by arorarshi** [Bioconductor/Contributions#2829]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/arorarshi/survClust
- 
Confirm the following by editing each check box to '[x]'

- [x] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [x] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [x] I understand Bioconductor [Package Naming Policy][9] and acknowledge
  Bioconductor may retain use of package name.

- [x] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [x] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [x] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.
  
- [x] I am familiar with the [Bioconductor code of conduct][7] and 
  agree to abide by it.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [x] The 'devel' branch for new packages and features.
- [x] The stable 'release' branch, made available every six
      months, for bug fixes.
- [x] _Bioconductor_ version control using [Git][5]
  (optionally [via GitHub][6]).

For questions/help about the submission process, including questions about
the output of the automatic reports generated by the SPB (Single Package
Builder), please use the #package-submission channel of our Community Slack.
Follow the link on the home page of the [Bioconductor website][8] to sign up.

[1]: https://contributions.bioconductor.org/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/git/
[6]: http://bioconductor.org/developers/how-to/git/sync-existing-repositories/
[7]: https://bioconductor.org/about/code-of-conduct/
[8]: https://bioconductor.org/
[9]: https://bioconductor.org/developers/package-submission/#naming

**Comment by vjcitn** [comment]
Thanks for this submission.  The example data is in a form that does not reflect Bioconductor practice.  You have a list of matrices or data frames.  We use SummarizedExperiment and derived classes to couple the metadata to data to increase reliability.  You should have a look at curatedTCGAData package to see how mutation and copynumber data can be represented in conventional ways for packages in the ecosystem.  Let me know @vjcitn how you want to proceed.

**Comment by arorarshi** [comment]
Hi @vjcitn, thanks for the feedback. survClust needs complex processing of genomic data as input and can't work with raw TCGA data. I am happy to work on the interop issue - is it ok if I convert the input matrixes (used as an example data to illustrate survClust) as a mae object? Will that work?  survClust takes mae objects as an input.

**Comment by arorarshi** [comment]
Hi. Thanks for getting back to me. I am now going through the report and fixing the errors. Thanks!

**Comment by Kayla-Morrell** [comment]
Hello @arorarshi - Checking in to see if you are making progress fixing the errors in the report. We generally like to see changes within 2-3 weeks to be sure the package is actively being worked on. Let me know if you are having any issues.

Best,
Kayla

**Comment by arorarshi** [comment]
Hi @Kayla-Morrell, Thanks for following up. Yes, I am working on fixing these errors today. Thanks!

**Comment by Kayla-Morrell** [comment]
@arorarshi - A week has gone by since you said you were working on fixing the errors. If changes are not made in the next week I will be closing the issue due to inactivity.

Best,
Kayla

**Comment by arorarshi** [comment]
Hi @Kayla-Morrell - I have resolved most of the errors except the bioc-devel mailing list. I just subscribed to it but looks like it might need approval? Should I try to commit after the confirmation that I have been added to the list?

**Comment by lshep** [comment]
I don't see you registered. There is normally a confirmation email that gets sent to your email and sometimes it can be marked as spam. Could you please check your spam filter

**Comment by Kayla-Morrell** [comment]
@arorarshi - Thank you for submitting to _Bioconductor_. Please see the initial review of
the package below. The required changes must be made while the suggested
changes do not have to be (though we strongly encourage them). Comment back
here with updates that have been made and when the package is ready for a
re-review.

## DESCRIPTION

- [ ] REQUIRED: The LICENSE file should contain the full terms of the MIT
  license. I would remove the current LICENSE file and rename LICENSE.md as
LICENSE.
- [ ] REQUIRED: 'LazyData:' in the 'DESCRIPTION' should be set to false or
  removed.
- [ ] REQUIRED: Update R version dependency from 4.2.0 to 4.3.0.
- [ ] REQUIRED: There are packages used in vignette code that should be declared
  in the `Suggests:` field. Please add 'gplots', 'htmltools', and
'BiocParallel'.

## NAMESPACE

- [ ] REQUIRED: Exported functions should use camel case or underscoring and not
  include `.`, which indicates S3 dispatch. Please change 'cv.survclust'.
- [ ] SUGGESTION: Add the follwing to your NAMESPACE file and ensure that your
  DESCRIPTION Imports field contains these packages.
        importFrom("graphics", "axis", "boxplot", "par")
        importFrom("methods", "is")
        importFrom("stats", "cmdscale", "dist", "kmeans")

## Vignette

- [ ] SUGGESTION: Change the vignette file name to something pertaining to the
  package, such as 'survClust', 'survClust_vignette', or 'survClust_intro'.
- [ ] REQUIRED: All code blocks should be evaluated, except for installation
  code. If it is necessary please provide justification.
- [ ] REQUIRED: There should be an Installation section that demonstrates to the
  user how to download and load the package from Bioconductor. The code should
look like the following and include `eval = FALSE`.

```{r}
if(!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("survClust")
```
- [ ] SUGGESTION: I would remove one '#' from each of the headers and
  subheadings. Right now the table of contents reads '0.1 Overview', '0.1.1 The
tldr version', '0.2 Data and Pre-processing', etc. If you removed one '#' from
each then they should become '1 Overview', '1.1 The tldr version', '2 Data and
Pre-processing', etc.

## Man pages

- [ ] REQUIRED: There is no documenation for the exported function
  'dist_wtbinary'. All exported functions must have a man page.
- [ ] REQUIRED: Add a runnable example to the 'get_spwss_stats' man page since
  this documents an exported function.

## R code

- [ ] REQUIRED: Line 35 in 'combineDist.R', `dim` in the `rowMeans` function
  should be `dims`.
- [ ] REQUIRED: Avoid `1:...`; use `seq_len()` or `seq_along()` instead.
- [ ] SUGGESTION: For formating reasons, consider shorter lines. There are 214
  lines that are > 80 characters long.
- [ ] SUGGESTION: For formating reasons, consider multiples of 4 spaces for line
  indents. There are 14 lines that are not.

Best,
Kayla

**Comment by arorarshi** [comment]
Dear Bioconductor team,I am starting my maternity leave starting today. I was able to work on most of these changes but will only be able to push them to Git when I am more settled. Will it be possible to request for more time to work on the package and get it ready for bioconductor submission? Any other hard deadlines that I should keep in mind?Thank you,Arshi On Feb 7, 2023, at 1:59 PM, Kayla-Morrell ***@***.***> wrote:﻿
@arorarshi - Thank you for submitting to Bioconductor. Please see the initial review of
the package below. The required changes must be made while the suggested
changes do not have to be (though we strongly encourage them). Comment back
here with updates that have been made and when the package is ready for a
re-review.
DESCRIPTION

 REQUIRED: The LICENSE file should contain the full terms of the MIT
license. I would remove the current LICENSE file and rename LICENSE.md as
LICENSE.
 REQUIRED: 'LazyData:' in the 'DESCRIPTION' should be set to false or
removed.
 REQUIRED: Update R version dependency from 4.2.0 to 4.3.0.
 REQUIRED: There are packages used in vignette code that should be declared
in the Suggests: field. Please add 'gplots', 'htmltools', and
'BiocParallel'.

NAMESPACE

 REQUIRED: Exported functions should use camel case or underscoring and not
include ., which indicates S3 dispatch. Please change 'cv.survclust'.
 SUGGESTION: Add the follwing to your NAMESPACE file and ensure that your
DESCRIPTION Imports field contains these packages.
importFrom("graphics", "axis", "boxplot", "par")
importFrom("methods", "is")
importFrom("stats", "cmdscale", "dist", "kmeans")

Vignette

 SUGGESTION: Change the vignette file name to something pertaining to the
package, such as 'survClust', 'survClust_vignette', or 'survClust_intro'.
 REQUIRED: All code blocks should be evaluated, except for installation
code. If it is necessary please provide justification.
 REQUIRED: There should be an Installation section that demonstrates to the
user how to download and load the package from Bioconductor. The code should
look like the following and include eval = FALSE.

if(!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("survClust")


 SUGGESTION: I would remove one '#' from each of the headers and
subheadings. Right now the table of contents reads '0.1 Overview', '0.1.1 The
tldr version', '0.2 Data and Pre-processing', etc. If you removed one '#' from
each then they should become '1 Overview', '1.1 The tldr version', '2 Data and
Pre-processing', etc.

Man pages

 REQUIRED: There is no documenation for the exported function
'dist_wtbinary'. All exported functions must have a man page.
 REQUIRED: Add a runnable example to the 'get_spwss_stats' man page since
this documents an exported function.

R code

 REQUIRED: Line 35 in 'combineDist.R', dim in the rowMeans function
should be dims.
 REQUIRED: Avoid 1:...; use seq_len() or seq_along() instead.
 SUGGESTION: For formating reasons, consider shorter lines. There are 214
lines that are > 80 characters long.
 SUGGESTION: For formating reasons, consider multiples of 4 spaces for line
indents. There are 14 lines that are not.

Best,
Kayla

**Comment by Kayla-Morrell** [comment]
@arorarshi - I'll close it for inactivity for now. When you are ready to make changes just comment on the issue and I'll reopen it for you. We will be releasing deadlines soon for the spring release but these are not hard deadlines. If you do not make the spring release your package will be eligible for the next fall release, so no rush on our end.

Best,
Kayla

**Comment by arorarshi** [comment]
Hi there, things are settling in after maternity leave, I am working on these changes and should be able to push soon. Should the issue be opened and then push the changes?

**Comment by lshep** [comment]
yes that way there is a new build report generated.  Cheers

**Comment by Kayla-Morrell** [comment]
@arorarshi - Thank you for making the necessary changes. I have reviewed everything and it all looks good. I'm more than happy to accept the package.

Best,
Kayla

**Comment by arorarshi** [comment]
Thank you! This is excellent news. This is the outcome of the Bioconductor package mentorship program in 2022 :) I was finding my footing after maternity leave and would like to give a shout out to all the reviewers and my mentors @DarioS @nturaga

**Comment by lshep** [comment]
The default branch of your GitHub repository has been added to Bioconductor's
git repository as branch devel.

To use the git.bioconductor.org repository, we need an 'ssh' key to associate with your github user name. If your GitHub account already has ssh public keys (https://github.com/arorarshi.keys is not empty), then no further steps are required. Otherwise, do the following:

1. [Add an SSH key to your github account][]
1. [Submit your SSH key to Bioconductor][]

See further instructions at

https://bioconductor.org/developers/how-to/git/

for working with this repository. See especially

https://bioconductor.org/developers/how-to/git/new-package-workflow/
https://bioconductor.org/developers/how-to/git/sync-existing-repositories/

to keep your GitHub and Bioconductor repositories in sync.

Your package will be included in the next nigthly 'devel' build (check-out from git at about 6 pm Eastern; build completion around 2pm Eastern the next day) at

https://bioconductor.org/checkResults/

(Builds sometimes fail, so ensure that the date stamps on the main landing page are consistent with the addition of your package). Once the package builds successfully, you package will be available for download in the 'Devel' version of Bioconductor using `BiocManager::install("survClust")`. The package 'landing page' will be created at

https://bioconductor.org/packages/survClust

If you have any questions, please contact the bioc-devel mailing list (https://stat.ethz.ch/mailman/listinfo/bioc-devel); this issue will not be monitored further.

[Add an SSH key to your github account]: https://help.github.com/articles/connecting-to-github-with-ssh/
[Submit your SSH key to Bioconductor]: https://git.bioconductor.org/BiocCredentials/
